en/README.md

```diff
@@ -0,0 +1,1138 @@
+|:-----------------|:--|:-:|:----------:|:---|
+|triggers    |object array|M|-|This is object array that stores triggers information. Look at the following table for details.|
+|updateType|string    |M|-|Select sending option that match the situation from [[List](#user-content-updatetype)].|
+|lastInfo    |String32767|O|-|Insert to here trigger information of last. This information is result of [getLastInfo](#user-content-getlastinfomethod).|
+|fetchId|String255|O|-|This is ID that indicates whether response for request from Hatohol server. Insert fetchId value to this object, in case of only receive fetchTriggers request.
+
+***triggers object***
+
+|Object name|Type |M/O|Default value|Description|
+|:-----------------|:--|:-:|:----------:|:---|
+|triggerId     |String255  |M|-|Id of the trigger. When send trigger of HAP itself, must insert "_SELF_" to trigger Id and host Id.|
+|status        |string     |M|-|Status of trigger [[List](#user-content-triggerstatus)].|
+|severity      |string     |M|-|Severity of trigger [[List](#user-content-triggerseverity)].|
+|lastChangeTime|TimeStamp  |M|-|Time of last updated trigger.|
+|hostId        |String255  |M|-|Id of host that is trigger of host belongs.|
+|hostName      |String255  |M|-|Host name of host that is trigger of host belongs.|
+|brief         |String255  |M|-|Overview of trigger.|
+|extendedInfo  |String32767|M|-|The extra information other than the followings. It uses to show data on WebUI mainly.|
+
+```json
+{
+  "id": 1,
+  "params": {
+    "triggers": [
+      {
+        "extendedInfo": "sample extended info",
+        "brief": "example brief",
+        "hostName": "exampleName",
+        "hostId": "1",
+        "lastChangeTime": "20150323175800",
+        "severity": "INFO",
+        "status": "OK",
+        "triggerId": "1"
+      }
+    ],
+    "fetchId": "1",
+    "lastInfo": "201504061606",
+    "updateType": "UPDATED"
+  },
+  "method": "putTriggers",
+  "jsonrpc": "2.0"
+}
+```
+
+***Result(result)***
+
+Receive result as result object value whether sent data has been updated. Look at the [[Values](#user-content-putresult)].
+
+```json
+{
+  "id": 1,
+  "result": "SUCCESS",
+  "jsonrpc": "2.0"
+}
+```
+
+### putEvents(method)
+
+putEvents procedures can send till 1000 events. If want to send over than 1000, send events to separate more than once.
+
+Can send event that duplicates event Id. But, to taken priority existing event or new event does not define.
+
+It has two behavior that are to send voluntary and send as response for fetchEvents.
+
+In case of voluntary, send diffarence from before events to use lastInfo that got using [getLastInfo](#user-content-getlastinfomethod) or saved in internal to Hatohol server. When HAP connect first time, lastInfo is nothing, so HAP producer can select to send all existing event or do nothing.
+
+In case of ahead event from to designate event is nothing, send events object as empty string. 
```

Comment:
>  In case of ahead event from to designate event is nothing, send events object as empty string. 

If there is no event earlier than the specified events, it should be send event object as empty string.